### DISCUSSION TITLE
Feature(s) Request: Select All, Scale, Rotate, export sketch

### DISCUSSION THREAD

**Discussion by Shaddy-ElDesouky** [hlorus/CAD_Sketcher#324] [Feature Requests]
Hello everybody, You are Great, thanks a Million for your efforts.
I've gone through the Roadmap and Issues before adding this:
-Select all: maybe the same shortcut as Blender, lasso selection would be appreciated.
-Move (on the roadmap), but scale and rotate too would be helpful.
-copy paste (also on the roadmap): what about paste transformed (from solvespace) move or scale or rotate, or alltogether at once.
-Export sketch to: -solvespace slvs (if possible), -SVG (blender has no exporter, I found an addon -bezier2svg- but will export the curve not the sketch with its constraints), -DXF (blender exports dxf from mesh not from curve and that blows the Idea of exporting).

My use case: I'm an Interior designer, 1-I need to show renders to the client(covered by Blender-Cycles) 2-I need to send files to the CNC Router (here is where I need to export clean SVG or DXF for the router shop to generate  .Gcode  3- I need to generate 2D shop drawings for manufacturing(covered by blenderbim or sverchok or by exporting to freecad and finish there).

**Comment by hlorus** [comment]
Thanks, generally i'd advise to keep suggestions in single discussions to stay on topic and let people upvote them.

- Select Shortcut: I aggree it's confusing, the thing here is that blender doesn't follow the standard which is present in operatingsystems. IMO this could be changed if others agree as well.
- Move: A constraint based workflow is quite different from normal blender modeling and i'm afraid that people that are new to the addon might just use their habits with the addon and ignore the constraints. Therefor i'd say that's a low prio.
- Export: +1

**Comment by SiboVG** [comment]
I agree that it's a bit confusing that the Blender select options do not apply to the sketch mode. Would be helpful if pressing the A-key would select/deselect all entities instead of selecting the arc tool.

**Comment by mhmatthewhugley** [comment]
Being able to rotate a sketch is absolutely essential.

**Comment by glsys** [comment]
Is anybody working on svg-export for Sketches? If not I may give it a try...
